Mathlib/Algebra/Invertible.lean

```diff
@@ -54,6 +54,25 @@ def something_one := something_that_needs_inverses 1
 end
 ```
 
+### Typeclass search vs. unification for `simp` lemmas
+
+Note that since typeclass search searches the local context first, an instance argument like
+`[Invertible a]` might sometimes be filled by a different term than the one we'd find by
+unification (i.e., the one that's used as an implicit argument to `⅟`).
+
+This can cause issues with `simp`. Therefore, some lemmas are duplicated, with the `@[simp]`
+versions using unification and the user-facing ones using typeclass search.
+
+Since unification can make backwards rewriting (e.g. `rw [← mylemma]`)
```

Comment:
presumably it's not only slow, but the backwards rewrite will probably just fail if the LHS contains an `Invertible` type-class that the RHS doesn't contain, right? (Or at least `simp [← mylemma]` will fail.)

Maybe you can reformulate that. Other than that, LGTM

bors d+